singer/src/main/java/com/pinterest/singer/writer/KafkaWriter.java

```diff
@@ -338,7 +364,25 @@ public void writeLogMessages(List<LogMessage> logMessages) throws LogStreamWrite
           // get the max write latency
           maxKafkaBatchWriteLatency = Math.max(maxKafkaBatchWriteLatency,
               result.getKafkaBatchWriteLatencyInMillis());
+
+          if (SingerSettings.loggingAuditClient != null) {
+            for (Map.Entry<Integer, LoggingAuditHeaders> entry : listOfHeadersMap.get(bucketIndex)
+                .entrySet()) {
+                Integer listIndex = entry.getKey();
+                LoggingAuditHeaders loggingAuditHeaders = entry.getValue();
+                long messageAcknowledgedTimestamp = -1;
+
+                if (result.getRecordMetadataList().size() != buckets.get(bucketIndex).size()){
+                  LOG.error("Number of ProducerRecord does not match the number of RecordMetadata");
+                } else {
+                  RecordMetadata recordMetadata =result.getRecordMetadataList().get(listIndex);
```

Comment:
Why does this need to be checked for every record? Isn't it sufficient to do this only once?

Reply:
No need. made change in the update of the PR.  only check once per partition (one bucket of buckets)